math/wxMaxima/Portfile

```diff
@@ -1,59 +1,110 @@
+    version         18.02-dev-20180616
+    github.setup    andrejv wxmaxima b32fed0d535c05d589d1f573d2c0f4f839180d18
+    checksums       rmd160  86fcf85ddae42e84884a708d55acff393152ed23 \
+                    sha256  e34256294c51a37d80947b92d3cd979456bf0c8ae53a21ce6cbab63d99143bf2 \
+                    size    12739940
+
+    patchfiles      patch-devel-src_Dirstructure.cpp.diff
+    post-patch {
+        reinplace -W ${worksrcpath} "s|@@PREFIX@@|${prefix}|g" \
+                                    src/Dirstructure.cpp
+    }
+
+    livecheck.type  none
 }
 
-depends_skip_archcheck  maxima
-configure.args  --enable-printing --with-wxdir=${wxWidgets.wxdir}
-build.target-append wxMaxima.app
+if {${subport} eq ${name}} {
+    conflicts       wxMaxima-devel
+    long_description ${description}: \
+                    Provides the release version, which is typically updated every four months.
 
-destroot {
-    xinstall -m 755 -d ${destroot}${applications_dir}
-    file copy ${worksrcpath}/wxMaxima.app ${destroot}${applications_dir}
+    # get the source tarball from github.
+    github.setup    andrejv wxmaxima 18.02.0 Version-
+    checksums       rmd160  d1017d56bfd76e9c51dcc3230be89ebca1724560 \
+                    sha256  77f0414be54d8ea15acedfc3cedf72a57cc2b3ba5d4835a53f3009e050c05483 \
+                    size    10131293
 
-    set docPath ${prefix}/share/doc/${name}
-    xinstall -m 755 -d ${destroot}${docPath}
-    foreach f { AUTHORS COPYING README } {
-        xinstall -m 644 ${worksrcpath}/${f} ${destroot}${docPath}
+    patchfiles      patch-src_Dirstructure.cpp.diff
+    post-patch {
+        reinplace -W ${worksrcpath} "s|@@PREFIX@@|${prefix}|g" \
+                                    src/Dirstructure.cpp
     }
+    
+    livecheck.regex Version-(\[a-z0-9.\]+)${extract.suffix}
 }
 
+universal_variant   no
+
+variant wx32 description {Use wxWidgets 3.2 - experimental!} {}
+
+if {[variant_exists wx32] && [variant_isset wx32]} {
+    # we want to use the wxWidgets-3.2
+    wxWidgets.use   wxWidgets-3.2
+} else {
+    wxWidgets.use   wxWidgets-3.0
+}
+
+depends_lib         port:${wxWidgets.port}
+
+depends_build       port:cmake
+
+depends_run         bin:maxima:maxima
+
+patch.pre_args      -Np0
+
+use_configure       no
+
+build {
+    system -W ${worksrcpath}       "mkdir ./build"
+    system -W ${worksrcpath}/build "cmake -DwxWidgets_CONFIG_EXECUTABLE:FILEPATH=${wxWidgets.wxconfig} .."
+    system -W ${worksrcpath}/build "cmake --build ."
+}
+
+destroot {
+    xinstall -m 755 -d ${destroot}${applications_dir}
+    file copy ${worksrcpath}/build/wxMaxima.app ${destroot}${applications_dir}
+}
 
+notes "
+    Be careful: \n\
+    If you want to use wxMaxima with default setting, \n\
+    you should enter the command \n\
+    \"open -a ${applications_dir}/wxMaxima.app\" \n\
+    on Terminal.app in order to launch wxMaxima. \n\
+    Because you must let wxMaxima know the path to softwares \n\
```

Comment:
nitpick: I'm pretty sure that the "softwares" is not a word. The plural of software is software.

Reply:
Oh, thank you. I will change the  'softwares' to 'software'.